### PR TITLE
Issue #473

### LINKED ISSUE #473: Log level filtering by --loglevel flag
**Is your feature request related to a problem? Please describe.**
We log everything to console and/or to the log file without filter
Also mentioned in https://github.com/pokt-network/pocket-core/issues/130

**Describe the solution you'd like**
A flag like --loglevel where the user can pass the kind of loglevel to log

### PR DESCRIPTION
**Pull request by aqt01**
Closes #473 #130 

log-errs

### CODE REVIEW

File: tests/unit/log_test.go [hunk 1]
```diff
@@ -9,16 +9,58 @@ import (
 	utils "github.com/pokt-network/pocket-core/tests/unit/utils"
 )
 
+func TestTextLogs(t *testing.T) {
+	if err := logs.Log("Unit test for the json log functionality", logs.InfoLevel, logs.TextLogFormatter); err != nil {
```
Comment (luyzdeleon) [inline tests/unit/log_test.go]: There's a typo in text, it says: `Unit test for the json log functionality` when it should say: `Unit test for the text log functionality`

File: tests/unit/log_test.go [hunk 2]
```diff
@@ -9,16 +9,58 @@ import (
...
 func TestJSONLogs(t *testing.T) {
 	if err := logs.Log("Unit test for the json log functionality", logs.InfoLevel, logs.JSONLogFormat); err != nil {
 		t.Fatalf(err.Error())
 	}
 }
 
-func TestTextLogs(t *testing.T) {
-	if err := logs.Log("Unit test for the text log functionality", logs.InfoLevel, logs.TextLogFormatter); err != nil {
+func TestJSONLogLevel(t *testing.T) {
+	if err := logs.Log("Unit test for the trace log functionality", logs.TraceLevel, logs.JSONLogFormat); err != nil {
```
Comment (luyzdeleon) [inline tests/unit/log_test.go]: The text should indicate the log type as well as the level, e.g.: `Log type JSON Log level TRACE`

File: tests/unit/log_test.go [hunk 3]
```diff
@@ -9,16 +9,58 @@ import (
...
+	}
+	if err := logs.Log("Unit test for the warn log functionality", logs.WarnLevel, logs.JSONLogFormat); err != nil {
+		t.Fatalf(err.Error())
+	}
+
+}
+
+func TestTextLogLevel(t *testing.T) {
+
+	if err := logs.Log("Unit test for the trace log functionality", logs.TraceLevel, logs.TextLogFormatter); err != nil {
```
Comment (luyzdeleon) [inline tests/unit/log_test.go]: See above.

File: tests/unit/log_test.go [hunk 4]
```diff
@@ -9,16 +9,58 @@ import (
...
+	if err := logs.Log("Unit test for the debug log functionality", logs.DebugLevel, logs.TextLogFormatter); err != nil {
+		t.Fatalf(err.Error())
+	}
+	if err := logs.Log("Unit test for the warn log functionality", logs.WarnLevel, logs.TextLogFormatter); err != nil {
 		t.Fatalf(err.Error())
 	}
+
+}
+
+func TestLoglevelFlag(t *testing.T) {
```
Comment (luyzdeleon) [inline tests/unit/log_test.go:51]: Should a test for when the incorrect log level is passed to the flag.

**Review by luyzdeleon — CHANGES_REQUESTED** [review]

**Review by andrewnguyen22 — APPROVED** [review]

**Pull request merged**